packages/common/src/utils/regex/RegexUtils.tsx

```diff
@@ -28,4 +28,19 @@ export const RegexUtils = {
     const matcher = new RegExp(substring, 'i');
     return matcher.test(testString);
   },
+  // Case-insensitive version of 'startsWith'
+  matchString: (substring: string, testString: string) => {
+    const matcher = new RegExp(`^${substring}`, 'i');
+    return matcher.test(testString);
+  },
+  // matches test string against an object's name or code properties
+  matchNameOrCode: (
+    item: { name: string; code: string },
+    searchString: string
+  ) => {
+    return (
+      RegexUtils.matchString(searchString, item.name) ||
+      RegexUtils.matchString(searchString, item.code)
```

Comment:
This is good, I've never been able to figure out how to refer to a value from another property of the same object! Thanks :)